primitives/src/lib.rs

```diff
@@ -477,17 +477,29 @@ create_currency_id! {
     }
 }
 
+#[derive(Encode, Decode, Eq, Hash, PartialEq, Copy, Clone, RuntimeDebug, PartialOrd, Ord, TypeInfo, MaxEncodedLen)]
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+#[cfg_attr(feature = "std", serde(rename_all = "camelCase"))]
+pub enum LpToken {
+    Token(TokenSymbol),
+    ForeignAsset(ForeignAssetId),
+    StableLpToken(StablePoolId),
+}
+
 #[derive(Encode, Decode, Eq, Hash, PartialEq, Copy, Clone, RuntimeDebug, PartialOrd, Ord, TypeInfo, MaxEncodedLen)]
 #[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
 #[cfg_attr(feature = "std", serde(rename_all = "camelCase"))]
 pub enum CurrencyId {
     Token(TokenSymbol),
     ForeignAsset(ForeignAssetId),
     LendToken(LendTokenId),
+    LpToken(LpToken, LpToken),
+    StableLpToken(StablePoolId),
```

Comment:
Why is a stableLpToken identified by a single id whereas normal lptokens consist of a pair? That seems inconsistent to me. I realize that it's probably dictated by the zenlink pallet, I'm just wondering

Reply:
Technically the stable pool can support multiple assets whereas the standard AMM can only ever support pairs. However, I see no reason why both couldn't be implemented in the same way but it is probably more effort than is worth it right now for Zenlink to adopt the same interface.